TeachingRecordSystem/src/TeachingRecordSystem.Core/Services/TrsDataSync/TrsDataSyncHelper.cs

```diff
@@ -37,10 +34,44 @@ public class TrsDataSyncHelper(
     };
 
     private readonly ISubject<object[]> _syncedEntitiesSubject = new Subject<object[]>();
+    private readonly NpgsqlDataSource _trsDbDataSource;
+    private readonly IOrganizationServiceAsync2 _organizationService;
+    private readonly ReferenceDataCache _referenceDataCache;
+    private readonly IClock _clock;
+    private readonly IAuditRepository _auditRepository;
+    private readonly ILogger<TrsDataSyncHelper> _logger;
+    private bool? isFakeXrm;
+
+    public TrsDataSyncHelper(
+        NpgsqlDataSource trsDbDataSource,
+        [FromKeyedServices(TrsDataSyncService.CrmClientName)] IOrganizationServiceAsync2 organizationService,
+        ReferenceDataCache referenceDataCache,
+        IClock clock,
+        IAuditRepository auditRepository,
+        ILoggerFactory loggerFactory)
```

Comment:
Can probably revert to a primary constructor with this too
```suggestion
        ILogger<TrsDataSyncHelper> logger)
```